NDTensors/src/lib/BlockSparseArrays/src/blocksparsearray/blockdiagonalarray.jl

```diff
@@ -0,0 +1,60 @@
+# type alias for block-diagonal
+using LinearAlgebra: Diagonal
+
+const BlockDiagonal{T,A,Axes,V<:AbstractVector{A}} = BlockSparseMatrix{
+  T,A,Diagonal{A,V},Axes
+}
+
+function BlockDiagonal(blocks::AbstractVector{<:AbstractMatrix})
+  return BlockSparseArray(
+    Diagonal(blocks), (blockedrange(size.(blocks, 1)), blockedrange(size.(blocks, 2)))
+  )
+end
+
+# Cast to block-diagonal implementation if permuted-blockdiagonal
+function find_permute_blockdiagonal(A)
```

Comment:
This doesn't quite fit how `find` is generally used in `Base` functions (i.e. it usually is used when a function returns the index of a certain value in a collection).

Functions in `Base` that return a permutation generally use `perm`, like `sortperm`. So maybe a better name would be `try_blockdiagonal_perm` or `maybe_blockdiagonal_perm` to indicate that it outputs a permutation if possible.

Reply:
Or maybe `try_make_blockdiagonal_perm` to match the function `try_make_blockdiagonal`.